atc/db/container_repository.go

```diff
@@ -362,3 +363,49 @@ func (repository *containerRepository) DestroyFailedContainers() (int, error) {
 
 	return int(failedContainersLen), nil
 }
+
+func (repository *containerRepository) DestroyUnknownContainers(workerName string, reportedHandles []string) (int, error) {
+	dbHandles, err := repository.queryContainerHandles(sq.Eq{
+		"worker_name": workerName,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	unknownHandles := diff(reportedHandles, dbHandles)
+
+	if len(unknownHandles) == 0 {
+		return 0, nil
+	}
+
+	tx, err := repository.conn.Begin()
```

Comment:
nit: Not something you need to change but you probably don't need to open a transaction anymore, can probably just use the connection in the query.